jwst/spectral_leak/spectral_leak.py

```diff
@@ -0,0 +1,49 @@
+import numpy as np
+import logging
+from stdatamodels.jwst import datamodels
+
+log = logging.getLogger(__name__)
+log.setLevel(logging.DEBUG)
+
+
+def do_correction(sp_leak_ref, ch1b, ch3a):
+    """
+    Short Summary
+    -------------
+    Using the spectral leak reference correction and spectrum containing CH1 B
+    correct the CH 3 A spectrum.
+
+    Parameters
+    ----------
+    sp_leak_ref: ~datamodels.MirMrsPtCorrModel
+        spectral leak reference information
+
+    ch1b: numpy array
+    ch3a: numpy array
+
+    Returns
+    -------
+    output_model: ~jwst.datamodels.RampModel
+        RSCD-corrected science data
+    """
+
+    wave1b = ch1b.spec[0].spec_table.WAVELENGTH
+    spec1b = ch1b.spec[0].spec_table.FLUX
+
+    wave3a = ch3a.spec[0].spec_table.WAVELENGTH
+    spec3a = ch3a.spec[0].spec_table.FLUX
+
+    leak_ref = datamodels.MirMrsPtCorrModel(sp_leak_ref)
+    leak_wave = leak_ref.leakcor_table.wavelength
+    leak_percent = leak_ref.leakcor_table.frac_leak
+
+    # Spectral leak vector
+
+    leak = np.interp(wave3a, 2*wave1b, spec1b) * np.interp(wave3a, leak_wave, leak_percent)
+    # Factor of 2 in 2*wavelb is  necessary because we're dealing with second order light.  Without this the interpolation simply
+    # uses the last point in the spectrum, which is close enough to correct as won't be noticeable for BAND spectra,
+    # but for CH spectra can be quite different.
```

Comment:
"Note that the factor of 2 used above in "2*wave1b" is necessary, because we're dealing with second-order light.
  Without this, the interpolation would simply use the last point in the spectrum, which would be close enough
  to not be noticeable for BAND spectra, but can be quite different for Channel spectra."